### ISSUE TITLE
Warn about invalid f-string specifiers

### ISSUE THREAD

**Issue by GideonBear** [astral-sh/ruff#25744]
### Summary

```py
subj = "world"
print(f"Hello {subj:r}")
```
```
Traceback (most recent call last):
  File "/home/gideon/delme/a.py", line 2, in <module>
    print(f"Hello {subj:r}")
                  ^^^^^^^^
ValueError: Unknown format code 'r' for object of type 'str'
```
This should be:
```diff
-print(f"Hello {subj:r}")
+print(f"Hello {subj!r}")
```

I suggest to either:
1. Warn about all unknown specifiers
2. Warn about unknown specifiers that have a corresponding conversion specifier (`s`, `r`, or `a`)

Downside: types can define their own specifier, so unless we know the type of `subj`, this can produce false positives.
I've personally never used that feature, and AFAICT it's pretty rarely used when compared to how often specifiers for `str`/`int` are used; so I think it can still be worth it to add it as an opt-in rule.

Inspired by https://github.com/jensjeflensje/marktplaats-py/pull/178

**Comment by MichaReiser** [comment]
I think the challenge here is that it requires knowing the type of `subj`, because each type can implement its own format codes

**Comment by jonathandung** [comment]
ty could consider the `__format__` method of the deduced type of the object in question, seeing if its `format_spec` parameter is typed as something like `Literal[<accepted format codes>]`, and then make a hint in the resultant diagnostic pointing to where `__format__` was declared if the format code used is not in that group.

**Comment by GideonBear** [comment]
> I think the challenge here is that it requires knowing the type of `subj`, because each type can implement its own format codes

It's not absolutely clear to me what rules ruff does and doesn't want to include; but IMO making a rule that only makes sense for projects that do not use custom format specifiers is still useful (clippy `restriction::` style).
